lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -221,11 +221,20 @@ public async Task<RevisionInfo> DownloadAsync(int revision)
 
             if (revisionInfo != null && GetCurrentPlatform() == Platform.Linux)
             {
-                int code = LinuxSysCall.Chmod(revisionInfo.ExecutablePath, LinuxSysCall.ExecutableFilePermissions);
+                string execPath = revisionInfo.ExecutablePath;
+                string dirName = Path.GetDirectoryName(execPath);
+
+                int code = LinuxSysCall.Chmod(execPath, LinuxSysCall.ExecutableFilePermissions);
                 if (code == -1)
                 {
                     throw new Exception("Chmod operation failed");
                 }
+
+                string naclPath = $"{dirName}/nacl_helper";
+                if (File.Exists(naclPath))
+                {
+                    LinuxSysCall.Chmod(naclPath, LinuxSysCall.ExecutableFilePermissions);
```

Comment:
We should check the code Chmod is returning, as we do in line 228

Reply:
done